src/ImageSharp.Web/Providers/PhysicalFileSystemProvider.cs

```diff
@@ -41,53 +28,23 @@ public class PhysicalFileSystemProvider : IImageProvider
             IWebHostEnvironment environment,
 #endif
             FormatUtilities formatUtilities)
+            : base(GetProvider(options?.Value, environment), formatUtilities)
         {
-            Guard.NotNull(options, nameof(options));
-            Guard.NotNull(environment, nameof(environment));
-
-            this.providerRootPath = GetProviderRoot(options.Value, environment.WebRootPath, environment.ContentRootPath);
-            this.formatUtilities = formatUtilities;
         }
 
         /// <inheritdoc/>
-        public ProcessingBehavior ProcessingBehavior { get; } = ProcessingBehavior.CommandOnly;
-
-        /// <inheritdoc/>
-        public Func<HttpContext, bool> Match { get; set; } = _ => true;
-
-        /// <inheritdoc/>
-        public bool IsValidRequest(HttpContext context)
-            => this.formatUtilities.TryGetExtensionFromUri(context.Request.GetDisplayUrl(), out _);
-
-        /// <inheritdoc/>
-        public Task<IImageResolver> GetAsync(HttpContext context)
-        {
-            // Use Join because request path starts with a slash
-            string fullPath = Path.GetFullPath(Path.Join(this.providerRootPath, context.Request.Path.Value));
-            if (PathUtilities.IsUnderneathRoot(fullPath, this.providerRootPath))
-            {
-                // Check to see if the file exists
-                var fileInfo = new FileInfo(fullPath);
-                if (fileInfo.Exists)
-                {
-                    var metadata = new ImageMetadata(fileInfo.LastWriteTimeUtc, fileInfo.Length);
-                    return Task.FromResult<IImageResolver>(new PhysicalFileSystemResolver(fileInfo, metadata));
-                }
-            }
-
-            return Task.FromResult<IImageResolver>(null);
-        }
+        public override ProcessingBehavior ProcessingBehavior { get; protected set; } = ProcessingBehavior.All;
```

Comment:
Why has this changed to `ProcessingBehavior.All`?
```suggestion
        public override ProcessingBehavior ProcessingBehavior { get; protected set; } = ProcessingBehavior.CommandOnly;
```

Reply:
Because the image could be stored in a folder that the static file middleware cannot touch. I’m still undecided about that and am thinking of making the setter public instead so users can use a factory to set it up

Reply:
Or even remove the setter and add the property to the options. That would be better actually…

Reply:
I've reverted the default since we default to wwwroot. Instead the value is passed via the options. We still require the protected set in order to set that value though unless we pass it via the constructor.

Reply:
Having it configurable for the `PhysicalFileSystemProvider` does indeed make sense if you've specified a provider root path that isn't inside the wwwroot (and therefore not served by the static file middleware).

Because the `WebRootImageProvider` is tightly coupled with the `WebRootFileProvider`, having that always use `ProcessingBehavior.CommandOnly` also makes sense 👍🏻